src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/EnumsShouldHaveZeroValue.cs

```diff
@@ -119,23 +119,30 @@ private static void AnalyzeSymbol(SymbolAnalysisContext context, INamedTypeSymbo
 
             if (symbol.HasAttribute(flagsAttribute))
             {
-                CheckFlags(symbol, zeroValuedFields, context.ReportDiagnostic);
+                var additionalEnumNoneNames = context.Options.GetStringOptionValue(
+                    EditorConfigOptionNames.AdditionalEnumNoneNames, RuleRename,
+                    symbol.Locations[0].SourceTree, context.Compilation)
+                    .Split(new[] { '|' }, StringSplitOptions.RemoveEmptyEntries)
+                    .ToImmutableArray();
+                CheckFlags(symbol, zeroValuedFields, additionalEnumNoneNames, context.ReportDiagnostic);
```

Comment:
Consider calculating `additionalEnumNoneNames` inside `CheckFlags` only for the case that uses it.

Reply:
Oh yeah, you are absolutely right! Thx